backends/graphics/gles-custom/gles-custom.cpp

```diff
@@ -0,0 +1,703 @@
+	if (!gbm.surface) {
+		debug ("failed to create gbm surface\n");
+		return -1;
+	}
+	
+	return 0;
+}
+
+#endif
+
+static struct {
+	EGLDisplay display;
+	EGLConfig config;
+	EGLContext context;
+	EGLSurface surface;
+	uint32_t width, height, refresh;
+} eglInfo;
+
+void OpenGLCustomGraphicsManager::init_egl(){
+	static const EGLint attribute_list[] =
+	{
+	    EGL_RED_SIZE, 8,
+	    EGL_GREEN_SIZE, 8,
+	    EGL_BLUE_SIZE, 8,
+	    EGL_ALPHA_SIZE, 8,
+	    EGL_SURFACE_TYPE, EGL_WINDOW_BIT,
+	    EGL_NONE
+	};
+	
+	// create an EGL rendering context
+	static const EGLint context_attributes[] =
+	{
+	    EGL_CONTEXT_CLIENT_VERSION, 1,
+	    EGL_NONE
+	};
+
+	EGLint num_config;
+	
+	#ifdef USE_GLES_RPI
+	bcm_host_init();
+	VC_RECT_T dst_rect;
+	VC_RECT_T src_rect;
+	
+	// create a Dispmanx EGL window surface
+	int32_t success = graphics_get_display_size(0, &eglInfo.width, &eglInfo.height);
+	assert(success >= 0);
+
+	vc_dispmanx_rect_set(&dst_rect, 0, 0, eglInfo.width, eglInfo.height);
+    
+	vc_dispmanx_rect_set(&src_rect, 0, 0, eglInfo.width << 16, eglInfo.height << 16);
+    
+	dispman_display = vc_dispmanx_display_open(0);
+	dispman_update = vc_dispmanx_update_start(0);
+	dispman_element = vc_dispmanx_element_add(dispman_update, dispman_display,
+                                              10, &dst_rect, 0, &src_rect,
+                                              DISPMANX_PROTECTION_NONE, NULL, NULL, DISPMANX_NO_ROTATE);
+    
+	nativewindow.element = dispman_element;
+	nativewindow.width = eglInfo.width;
+	nativewindow.height = eglInfo.height;
+    
+	vc_dispmanx_update_submit_sync(dispman_update);
+	#endif    
+	
+	#ifdef USE_GLES_FBDEV
+	struct fb_var_screeninfo vinfo;
+	int fb = open("/dev/fb0", O_RDWR, 0);
+	if (ioctl(fb, FBIOGET_VSCREENINFO, &vinfo) < 0) {
+		debug("Error obtainig framebuffer info\n");
+		return;
+	}
+	close (fb);
+
+	nativewindow.width = vinfo.xres;
+	nativewindow.height = vinfo.yres;
+	#endif	
+
+	#ifdef USE_GLES_KMS
+	if (init_drm()) {
+		debug("failed to initialize DRM\n");
+		return;
+	}
+
+	if (init_gbm()) {
+		debug("failed to initialize GBM\n");
+		return;
+	}
+	#endif	
+	
+	// get an EGL display connection
+        #ifdef USE_GLES_KMS
+		eglInfo.display = eglGetDisplay((NativeDisplayType)gbm.dev);
+	#else
+		eglInfo.display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+	#endif
+	assert(eglInfo.display != EGL_NO_DISPLAY);
+
+	// initialize the EGL display connection
+	EGLBoolean result = eglInitialize(eglInfo.display, NULL, NULL);
+	assert(EGL_FALSE != result);
+    
+	// get an appropriate EGL frame buffer configuration
+	result = eglChooseConfig(eglInfo.display, attribute_list, &eglInfo.config, 1, &num_config);
+	assert(EGL_FALSE != result);
+    
+	result = eglBindAPI(EGL_OPENGL_ES_API);
+	assert(EGL_FALSE != result);
+	
+	eglInfo.context = eglCreateContext(eglInfo.display, eglInfo.config, EGL_NO_CONTEXT, context_attributes);
+	assert(eglInfo.context != EGL_NO_CONTEXT);
+
+	#ifdef USE_GLES_KMS
```

Comment:
Again, precompiler directive should start from column 0, without extra indentation